src/umpire/ResourceManager.hpp

```diff
@@ -347,6 +348,7 @@ class ResourceManager {
   friend std::vector<util::AllocationRecord> get_allocator_records(Allocator);
   friend strategy::ZeroByteHandler;
   friend strategy::mixins::AllocateNull;
+  friend std::size_t get_total_memory_allocated();
```

Comment:
Add a method to the get the names, rather than having this be a `friend` and accessing the data structure directly